packages/adapters/txservice/src/aggregator.ts

```diff
@@ -678,25 +679,71 @@ export class RpcProviderAggregator {
     if (needsSigner) {
       this.checkSigner();
     }
+
     const errors: NxtpError[] = [];
-    const shuffledProviders = this.shuffleSyncedProviders();
-    for (const provider of shuffledProviders) {
-      try {
-        return await method(provider);
-      } catch (e: unknown) {
-        // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
-        // however, as it will return instantly if the error is already a NxtpError.
-        const error = parseError(e);
-        if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
-          // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
-          // the call - so we'll retry the call with a different provider (if available).
-          errors.push(error);
-        } else {
-          // e.g. a TransactionReverted, TransactionReplaced, etc.
-          throw error;
+    const handleError = (e: unknown) => {
+      // TODO: With the addition of SyncProvider, this parse call may be entirely redundant. Won't add any compute,
+      // however, as it will return instantly if the error is already a NxtpError.
+      const error = parseError(e);
+      if (error.type === ServerError.type || error.type === RpcError.type || error.type === StallTimeout.type) {
+        // If the method threw a StallTimeout, RpcError, or ServerError, that indicates a problem with the provider and not
+        // the call - so we'll retry the call with a different provider (if available).
+        errors.push(error);
+      } else {
+        // e.g. a TransactionReverted, TransactionReplaced, etc.
+        throw error;
+      }
+    };
+
+    const quorum = this.config.quorum ?? 1;
+    if (quorum > 1) {
+      // Consult ALL providers.
+      const results: (T | undefined)[] = await Promise.all(
```

Comment:
this will actually be problematic if the providers hang -- we should have a cutoff so these promises dont just wait forever

Reply:
That's what syncprovider is for!

It does this internally. Check out SyncProvider.send in packages/adapters/txservice/src/shared/provider.ts

Reply:
oh okay awesome!